client/src/components/DataEntry/Ballot.tsx

```diff
@@ -73,14 +140,98 @@ const Ballot: React.FC<IProps> = ({
     IBallotInterpretation[]
   >(contests.map(emptyInterpretation))
+  const { confirm, confirmProps } = useConfirm()
+
+  const interpretiationsRef = useRef<IBallotInterpretation[]>(
+    contests.map(emptyInterpretation)
+  )
+  interpretiationsRef.current = interpretations
+
+  const renderInterpretation = (
+    { interpretation, choiceIds }: IBallotInterpretation,
+    contest: IContest
+  ) => {
+    if (!interpretation) return <div />
+    switch (interpretation) {
+      case Interpretation.VOTE:
+        return contest.choices.map(choice =>
+          choiceIds.includes(choice.id) ? (
+            <h3 key={choice.id}>{choice.name}</h3>
+          ) : null
+        )
+      case Interpretation.BLANK:
+        return <h3>Blank vote</h3>
+      case Interpretation.CONTEST_NOT_ON_BALLOT:
+        return <h3>Not on Ballot</h3>
+      // case Interpretation.CANT_AGREE:
+      // case for Interpretation.CANT_AGREE in case we decide to put it back in again
+      // return (
+      //   <LockedButton disabled large intent="primary">
+      //     Audit board can&apos;t agree
+      //   </LockedButton>
+      // )
+      default:
+        return null
+    }
+  }
+
+  const ballotSubmitFunc = (selectionType: 'not_found' | 'general') =>
+    selectionType === 'not_found'
+      ? submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
+      : submitBallot(
+          ballot.id,
+          BallotStatus.AUDITED,
+          interpretiationsRef.current.filter(
+            ({ interpretation }) => interpretation !== null
+          )
+        )
+
+  const initiateConfirm = (selectionType: 'not_found' | 'general') =>
+    confirm({
+      title: 'Confirm the Ballot Selections',
+      description: (
+        <>
+          {contests.map((contest, i) => (
+            <div key={contest.id}>
+              <p>{contest.name}</p>
+              {selectionType === 'not_found' ? (
+                <h3>Ballot Not Found</h3>
+              ) : (
+                <>
+                  {renderInterpretation(
+                    interpretiationsRef.current[i],
+                    contest
+                  )}
+                  <p>
+                    {interpretiationsRef.current[i].comment &&
+                      `Comment: ${interpretiationsRef.current[i].comment}`}
+                  </p>
+                </>
+              )}
+            </div>
+          ))}
+        </>
+      ),
+      onYesClick: async () => {
+        ballotSubmitFunc(selectionType)
+        nextBallot()
+      },
+      yesButtonLabel: 'Confirm Selections',
+      noButtonLabel: 'Change Selections',
+    })
+
+  const setInterpretationsFunc = (
+    newInterpretations: IBallotInterpretation[]
+  ) => {
+    setInterpretations(newInterpretations)
+    initiateConfirm('general')
+  }
 
   const submitNotFound = async () => {
-    await submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
-    nextBallot()
+    initiateConfirm('not_found')
```

Comment:
i think this whole section will be simpler if you have two separate functions for showing the confirm dialog instead of one function that conditions on general/not_found.

```ts
const onNotFoundClick = () =>
  confirm({
     ... all the confirm options for not found ...
     onYesClick: async () => {
      submitBallot(ballot.id, BallotStatus.NOT_FOUND, [])
      nextBallot()
     }
  })

const onSubmitSelectionsClick = async () =>
  confirm({
    ... all the confirm options for when interpretations were selected ...
    onYesClick: async () => {
      await submitBallot(
        ballot.id,
        BallotStatus.AUDITED,
        interpretations.filter(
          ({ interpretation }) => interpretation !== null
        )
      )
      nextBallot()
    }
  })